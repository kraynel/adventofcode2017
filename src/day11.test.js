const { countSteps, reduceSteps, countDistance } = require("./day11");
const fs = require("fs");
const _ = require("lodash");

describe("countSteps", () => {
  it("should return the total of each step", () => {
    expect(countSteps("ne,ne,ne")).toEqual({
      n: 0,
      s: 0,
      ne: 3,
      nw: 0,
      se: 0,
      sw: 0
    });
    expect(countSteps("ne,ne,sw,sw")).toEqual({
      n: 0,
      s: 0,
      ne: 2,
      nw: 0,
      se: 0,
      sw: 2
    });
  });

  it("should get the value", done => {
    fs.readFile("inputs/day11.txt", "utf8", (err, data) => {
      const steps = countSteps(data);
      expect(steps).toMatchSnapshot();
      expect(reduceSteps(steps)).toMatchSnapshot();
      expect(countDistance(steps)).toMatchSnapshot();

      const cumulativeSteps = {
        n: 0,
        s: 0,
        ne: 0,
        nw: 0,
        se: 0,
        sw: 0
      };
      const distances = data.split(",").map(value => {
        cumulativeSteps[value]++;
        return countDistance(cumulativeSteps);
      });
      expect(_.max(distances)).toMatchSnapshot();
      done();
    });
  });
});
