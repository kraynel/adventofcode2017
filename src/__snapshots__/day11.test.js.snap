// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countSteps should get the value 1`] = `
Object {
  "n": 1330,
  "ne": 1049,
  "nw": 1339,
  "s": 1697,
  "se": 1330,
  "sw": 1478,
}
`;

exports[`countSteps should get the value 2`] = `
Object {
  "n": 0,
  "ne": 0,
  "nw": 9,
  "s": 367,
  "se": 0,
  "sw": 429,
}
`;

exports[`countSteps should get the value 3`] = `796`;

exports[`countSteps should get the value 4`] = `1585`;
